### PR TITLE
chore: Remove usage of overlap hook in container

### PR DESCRIPTION
**Pull request by fralongo**
### Description

No longer needed after the removal of high-contrast header in these components

Related links, issue #, if available: n/a

### How has this been tested?

dev pipeline

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._ N/A
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._ N/A
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._ N/A
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._ N/A

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._ N/A

#### Testing

- _Changes are covered with new/existing unit tests?_ Y
- _Changes are covered with new/existing integration tests?_ Y
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/container/internal.tsx [hunk 1]
```diff
@@ -86,39 +85,20 @@ export default function InternalContainer({
...
-        setHasStickyBackground(false);
-      }
-    };
-  }, [isSticky, setHasStickyBackground, variant]);
+  const headerMergedRef = useMergeRefs(headerRef, __headerRef);
 
   // The container is only sticky on mobile if it is the header for the table.
   // In this case we don't want the container to have sticky styles, as only the table header row will show as stuck on scroll.
   const shouldHaveStickyStyles = isSticky && !isMobile;
+  const headerCoverHeight = stickyStyles?.style?.top ? stickyStyles.style.top : undefined;
```
Comment (georgylobko) [inline src/container/internal.tsx:101]:
```suggestion
  const headerCoverHeight = stickyStyles?.style?.top;
```

**Review by georgylobko — APPROVED** [review]
Looks good to me. However, left one optional comment

File: src/table/internal.tsx [hunk 2]
```diff
@@ -349,10 +346,11 @@ const InternalTable = React.forwardRef(
             __internalRootRef={__internalRootRef}
             className={clsx(baseProps.className, styles.root)}
             __funnelSubStepProps={__funnelSubStepProps}
+            __fullPage={variant === 'full-page'}
```
Comment (just-boris) [inline src/table/internal.tsx:349]: It seems like we already pass down the variant: https://github.com/cloudscape-design/components/blob/a247ab2ac7f005f8298e1eede66a991fc53bba81/src/table/internal.tsx#L390

Do we need another one?

**Review by just-boris — DISMISSED** [review]

File: src/container/use-sticky-header.ts [hunk 3]
```diff
@@ -42,7 +45,8 @@ export const useStickyHeader = (
   __stickyHeader?: boolean,
   __stickyOffset?: number,
   __mobileStickyOffset?: number,
-  __disableMobile = true
+  __disableMobile = true,
```
Comment (just-boris) [inline src/container/use-sticky-header.ts:48]: It seems like code coverage complains about this line. `__disableMobile` is never `undefined`, so it does not see coverage for the `= true` default. 

Simplest solution is to remove this default, it is not needed

**Review by just-boris — APPROVED** [review]

**Pull request merged**